src/io/ioexternalbus.cpp

```diff
@@ -28,10 +50,217 @@ QString IOExternalBus::description() const {
 }
 
 VInt IOExternalBus::ioRead(AInt offset, unsigned size) {
-    return 0;
+    uint32_t rvalue = 0;
+
+    if (tcpSocket->isOpen()) {
+        while (skt_use)
+            QApplication::processEvents();  // TODO only need in rvss, change to mutex
```

Comment:
I see this is marked as todo, but commenting just for good measure 👍. If a component needs processor/ISA specific information, this should be retrieved through the ProcessorInterface, and never through special-case code.

Reply:
Apparently only rvss tries to do reads and writes simultaneously, to use the socket, one operation has to finish to start another. I used a mutex to protect the socket and solve the problem. It is not architecture specific, but resource usage.